pinot-core/src/main/java/com/linkedin/pinot/core/query/executor/ServerQueryExecutorV1Impl.java

```diff
@@ -221,6 +233,46 @@ private long pruneSegments(TableDataManager tableDataManager, List<SegmentDataMa
     return totalRawDocs;
   }
 
+  /**
+   * Helper method to compute missing segments
+   *
+   * @param segmentsToQuery segments that need to be queried according to server query request
+   * @param segmentDataManagers acquired segment data managers from table data manager
+   * @return a list of missing segments
+   */
+  @Nonnull
+  private Set<String> computeMissingSegments(List<String> segmentsToQuery,
+      List<SegmentDataManager> segmentDataManagers) {
+    Set<String> missingSegments = new HashSet<>(segmentsToQuery);
+    for (SegmentDataManager segmentDataManager : segmentDataManagers) {
+      String segmentName = segmentDataManager.getSegmentName();
+      if (missingSegments.contains(segmentName)) {
```

Comment:
Can be simplified by directly calling missingSegments.remove(segmentName)

Reply:
fixed